src/Numerics/DGMethods/DGFVModel_kernels.jl

```diff
@@ -0,0 +1,800 @@
+        # the fluxes, so the minimum stencil diameter is 2
+        stencil_diameter = max(2, 2stencil_width + 1)
+
+        # Value in the stencil that corresponds to the top face with respect to
+        # face being updated
+        stencil_center = max(stencil_width, 1) + 1
+
+        # 1 → element i, face i - 1/2 (bottom face)
+        # 2 → element i, face i + 1/2 (top face)
+        local_state_face_prognostic = ntuple(Val(2)) do _
+            MArray{Tuple{num_state_prognostic}, FT}(undef)
+        end
+
+        local_cell_weights = MArray{Tuple{stencil_diameter}, FT}(undef)
+
+        # Two mass matrix inverse corresponding to +/- cells
+        vMI = MArray{Tuple{2}, FT}(undef)
+
+        # Storing the value below element when walking up the stack
+        # cell i-1, face i - 1/2
+        local_state_face_prognostic_neighbor =
+            MArray{Tuple{num_state_prognostic}, FT}(undef)
+
+        local_state_face_primitive = ntuple(Val(2)) do _
+            MArray{Tuple{num_state_primitive}, FT}(undef)
+        end
+
+        # Storage for all the values in the stencil
+        local_state_prognostic = ntuple(Val(stencil_diameter)) do _
+            MArray{Tuple{num_state_prognostic}, FT}(undef)
+        end
+
+
+        # Need to wrap in SVector so we can use views later
+        local_state_primitive = SVector(ntuple(Val(stencil_diameter)) do _
+            MArray{Tuple{num_state_primitive}, FT}(undef)
+        end...)
+
+        local_state_auxiliary = ntuple(Val(stencil_diameter)) do _
+            MArray{Tuple{num_state_auxiliary}, FT}(undef)
+        end
+
+        # FIXME: These two arrays could be smaller
+        # (only 2 elements not stencil_diameter)
+        local_state_gradient_flux = ntuple(Val(stencil_diameter)) do _
+            MArray{Tuple{num_state_gradient_flux}, FT}(undef)
+        end
+
+        local_state_hyperdiffusive = ntuple(Val(stencil_diameter)) do _
+            MArray{Tuple{num_state_hyperdiffusive}, FT}(undef)
+        end
+
+        # Storage for the numerical flux
+        local_flux = MArray{Tuple{num_state_prognostic}, FT}(undef)
+
+        # Storage for the extra boundary points for some BCs
+        local_state_prognostic_bottom1 =
+            MArray{Tuple{num_state_prognostic}, FT}(undef)
+        local_state_gradient_flux_bottom1 =
+            MArray{Tuple{num_state_gradient_flux}, FT}(undef)
+        local_state_auxiliary_bottom1 =
+            MArray{Tuple{num_state_auxiliary}, FT}(undef)
+
+        # XXX: will revisit this later for FVM
+        fill!(local_state_prognostic_bottom1, NaN)
+        fill!(local_state_gradient_flux_bottom1, NaN)
+        fill!(local_state_auxiliary_bottom1, NaN)
+
+        # The remainder model needs to know which direction of face the model is
+        # being evaluated for. In this case we only have `VerticalDirection()`
+        # faces
+        face_direction = (VerticalDirection())
```

Comment:
Based on downstream calls,
```julia
    ::Dirs,
) where {NumSubFluxes, S, A, Dirs <: NTuple{2, Direction}}
```
It looks like this should be a tuple: `face_direction = (VerticalDirection(), )`.

Reply:
Good catch, I dropped the comma it looks like.